### PR TITLE
Add a log function for handling complex values to the WebView client.

### LINKED ISSUE #4097: Element Call Widget Log consumer can't handle objects in log lines
### Steps to reproduce

1. Open Element X
2. Open a Element Call with someone
3. Observe the "[WebView]" logs in Element X.

#### What did you expect?

```
WARN elementx: [WebView] https://<domain>/assets/index-DyI8hxRz.js:56 websocket closed { details: "foo" }| 
```

#### What happened instead?

```
WARN elementx: [WebView] https://<domain>/assets/index-DyI8hxRz.js:56 websocket closed [object Object] |
```

### Your phone model

N/A

### Operating system version

N/A

### Application version and app store

v0.7.6

### Homeserver

N/A

### Will you send logs?

Yes

### Are you willing to provide a PR?

No

### PR DESCRIPTION
**Pull request by Half-Shot**
Fixes #4097 

This seems like the best way to achieve better logging, and apparently what the developers of Chromium expect people to do. https://issues.chromium.org/issues/41170324#comment4

### CODE REVIEW

**Comment by Half-Shot** [comment]
I'm not quite sure how we'd test a log function here, or whether a -0.02% loss of coverage is enough to block on this.

**Review by bmarty — APPROVED** [review]
LGTM and tested OK. Thanks!

File: features/call/impl/src/main/kotlin/io/element/android/features/call/impl/utils/WebViewWidgetMessageInterceptor.kt [hunk 1]
```diff
@@ -41,6 +41,27 @@ class WebViewWidgetMessageInterceptor(
             override fun onPageStarted(view: WebView?, url: String?, favicon: Bitmap?) {
                 super.onPageStarted(view, url, favicon)
 
+                // Due to https://github.com/element-hq/element-x-android/issues/4097
+                // we need to supply a logging implementation that correctly includes
+                // objects in log lines.
+                view?.evaluateJavascript(
+                    """
+                        function logFn(logFn, ...args) {
+                            logFn(
```
[the remark this answers is not in the document]
Comment (Half-Shot) [reply features/call/impl/src/main/kotlin/io/element/android/features/call/impl/utils/WebViewWidgetMessageInterceptor.kt:50]: that would make a ton of sense, yes.

[the remark this answers is not in the document]
Comment (bmarty) [reply features/call/impl/src/main/kotlin/io/element/android/features/call/impl/utils/WebViewWidgetMessageInterceptor.kt]: Thanks!

**Pull request merged**